WebDevStudios/constant-contact-forms
# Feature/cc 160 wcag compliance

Ticket: [CC-160](https://webdevstudios.atlassian.net/browse/CC-160)

### Description ###

Updates frontend form display for WCAG compliance:
- Removes IDs and placeholders from hidden fields.
- Adds `tabindex="-1"` to hidden fields.
- Adds `tabindex="-1"` to honeypot field.
- Updates "Phone Number" field to use `type="tel"` and "Website" field to use `type="url"`.
- Removes timeout of Ajax submission message to make it persistent.
- Adds `role` attribute to submission messages (both Ajax and standard).
- Add `ctct-` prefixed class to non-Ajax submission messages to match styling of Ajax messages.
- Adds 'X' button to Ajax notices to dismiss notice.

### Screenshot ###

![Screen Shot 2020-07-01 at 12 45 02 PM](https://user-images.githubusercontent.com/36422618/86280297-c0a9d480-bb98-11ea-8cc0-a08dcef9c183.png)

### Steps to verify the solution ###

1. Inspect form and confirm hidden fields no longer contain IDs or placeholders, but _do_ contain `tabindex="-1"`.
2. Tab through form and confirm "focus trap" on honeypot no longer exists.
3. Confirm "Phone Number" and "Website" fields are using the appropriate type. "Website" should auto-validate URLs (requiring full, absolute URLs).
4. Upon submission, error/success message should be color-coded (per `ctct-` prefixed classes) and contain the appropriate `role` attribute value.
5. Upon Ajax submission, error/success message should _not_ disappear after a few seconds. If a new message for the same form appears, previous messages should disappear at that time. Click the dismiss button ('X') and confirm the current message disappears.
